hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/Invocation.java

```diff
@@ -474,7 +475,7 @@ boolean detectAndHandleBackupTimeout(long timeoutMillis) {
         }
 
         if (shouldFailOnIndeterminateOperationState()) {
-            complete(new IndeterminateOperationStateException(this + " failed because backup acks missed."));
+            notifyError(new IndeterminateOperationStateException(this + " failed because backup acks missed."));
```

Comment:
Here we want to complete immediately with `IndeterminateOperationStateException `, without checking for any other condition.  I think `complete(..)` shows that intention better.

Reply:
:+1: